src/nvim/terminal.c

```diff
@@ -359,11 +359,13 @@ void terminal_enter(void)
   // Disable these options in terminal-mode. They are nonsense because cursor is
   // placed at end of buffer to "follow" output. #11072
   win_T *save_curwin = curwin;
-  int save_w_p_cul = curwin->w_p_cul;
```

Comment:
Let's keep this line, since if `'cursorlineopt'` equals e.g. `"line"` then we need to disable `cursorline` altogether. We will just restore it later in any case. (Although you can change `int` to `bool`).